src/modules/send_message/index.js

```diff
@@ -0,0 +1,74 @@
+const twitch = require('../../utils/twitch');
+const watcher = require('../../watcher');
+const debug = require('../../utils/debug');
+
+const chatCommands = require('../chat_commands');
+const anonChat = require('../anon_chat');
+const socketClient = require('../../socket-client');
+
+const PATCHED_SENTINEL = () => {};
+
+class SendState {
+    constructor(msg) {
+        this.message = msg;
+        this.defaultPrevented = false;
+    }
+
+    get user() {
+        return twitch.getCurrentUser();
+    }
+
+    preventDefault() {
+        this.defaultPrevented = true;
+        this.message = '';
+    }
+}
+
+let twitchSendMessage;
+const methodList = [
+    // msgObj => tabCompletion.onSendMessage(msgObj),
+    msgObj => chatCommands.onSendMessage(msgObj),
+    msgObj => anonChat.onSendMessage(msgObj)
+];
```

Comment:
emojis should be readded too.. it should be pretty much drop-in since it's mutating outgoing message